platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -295,6 +295,12 @@ plugins {
 
 We strongly encourage users to adopt the latest released version of the https://plugins.gradle.org/plugin/com.gradle.develocity[Develocity plugin], even when using it with older versions of Gradle.
 
+==== C++ and Swift plugins no longer depend on software model based plugins
+
+<<cpp_application_plugin.adoc#cpp_application_plugin,Cpp Application Plugin>>, <<cpp_library_plugin.adoc#cpp_library_plugin,Cpp Library Plugin>>, <<swift_application_plugin.adoc#swift_application_plugin,Swift Application Plugin>>, and <<swift_library_plugin.adoc#swift_library_plugin,Swift Library Plugin>> have been updated to no longer depend on software model plugin infrastructure.
+
+The impact of this change is that `toolChains` should be configured directly at the top-level of a build script and not inside a `model { }` block.
```

Comment:
```suggestion
As a result, `toolChains` should now be configured directly at the top-level of your build script instead of within a `model { }` block.
```